stumpy/core.py

```diff
@@ -943,19 +945,25 @@ def _calculate_squared_distance(m, QT, μ_Q, σ_Q, M_T, Σ_T):
         Window size
 
     QT : float
-        Dot product between `Q[i]` and `T[i]`
+        Pre-computed dot product between `Q` and `T`, each with length `m`.
```

Comment:
Remove period at the end

Reply:
> Hmm, I still feel like we should say `T[i]` since, generally,`T` is the entire time series

My only concern is the inconsistency between the meaning of variables. If we assume `T` is the entire time series, then `M_T` may be interpreted as sliding mean. But, if we just look at this function on its own (and forget about other functions), we should not talk about index `i` . 

IMHO: It is hard for me to read "the ith subsequence in `T`", and understand that `T` in QT or M_T corresponds to `T[i]`

Another perspective:
distance between two subseqs is symmetric. Hence, we may even avoid `Q` and `T`, and we can just say:

```
_calculate_squared_distance(m, S_AB, μ_SA, σ_SA, μ_SB, σ_SB)  # distance between `S_A` and `S_B`
```

The name I chose here is not good but I just want to say that we do not need to talk about query `Q` or `T` anymore. This is just a distance between two subsequences. So, if we mention `Q` and `T[i]`, it might sound like that this function can only compute distance between query `Q` and subseq `T[i]`. However, I think we should reflect that this function is general and it calculates the normalized ED between any two subsequences with the same length `m`.

Reply:
> I don't disagree. However, the algorithm for computing the distance is very much following the matrix profile procedure and it isn't so obvious why the distance is being computed in this way. I think it is better to keep the relationship to matrix profile by maintaining the variable names

That is reasonable! 

> How would you feel if we changed the arguments to something like:
> 
> ```
> _calculate_squared_distance(m, Q_T_i, μ_Q, σ_Q, M_T_i, Σ_T_i)
> ```

Tbh, I am not sure :) Since we do not have `i` in our code, it might sound weird. I don't know.... We may revise the description of the function to reflect the purpose of function better. 

(Btw, I should also mention that I might have been overthinking this since this function is private and we may just leave its description as is)

Reply:
> Since we do not have i in our code, it might sound weird

But the descriptions for each of the variables would say something about `T[i]` 

> (Btw, I should also mention that I might have been overthinking this since this function is private and we may just leave its description as is)

Yeah, let's just keep what you've changed so far and move on.